docs/notebook-demos/core-concepts.ipynb

```diff
@@ -0,0 +1,418 @@
+   "source": [
+    "gps.SpatialKey(col=0, row=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SpaceTimeKey\n",
+    "\n",
+    "Like `SpatialKey`s, `SpaceTimeKey`s describe the position of a raster in a layout. However, the grid is a 3D plane where a location of a raster is represented by a pair of coordinates, `col` and `row`, as well as a z value that represents a point in time called, `instant`. Like the `instant` in `TemporalProjectedExtent`, this is also an instance of `datetime.datetime`. Thus, `SpaceTimeKey`s deal with spatial-temporal data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "gps.SpaceTimeKey(col=0, row=0, instant=time)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Bounds\n",
+    "\n",
+    "`Bounds` represents the the extent of the layout grid in terms of keys. It has both a `minKey` and a `maxKey` attributes. These can either be a `SpatialKey` or a `SpaceTimeKey` depending on the type of data within the layer. The `minKey` is the left, uppermost cell in the grid and the `maxKey` is the right, bottommost cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Creating a Bounds from SpatialKeys\n",
+    "\n",
+    "min_spatial_key = gps.SpatialKey(0, 0)\n",
+    "max_spatial_key = gps.SpatialKey(10, 10)\n",
+    "\n",
+    "bounds = gps.Bounds(min_spatial_key, max_spatial_key)\n",
+    "bounds"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Creating a Bounds from SpaceTimeKeys\n",
+    "\n",
+    "min_space_time_key = gps.SpaceTimeKey(0, 0, 1.0)\n",
+    "max_space_time_key = gps.SpaceTimeKey(10, 10, 1.0)\n",
+    "\n",
+    "gps.Bounds(min_space_time_key, max_space_time_key)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Metadata\n",
+    "\n",
+    "`Metadata` contains information of the values within a layer. This data pertains to the layout, projection, and extent of the data contained within the layer.\n",
+    "\n",
+    "The below example shows how to construct `Metadata` by hand, however, this is almost never required and `Metadata` can be produced using easier means. For `RasterLayer`, one call the method, `collect_metadata()` and `TiledRasterLayer` has the attribute, `layer_metadata`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Creates Metadata for a layer with rasters that have a cell type of int16 with the previously defined\n",
+    "# bounds, crs, extent, and layout definition.\n",
+    "gps.Metadata(bounds=bounds,\n",
```

Comment:
Hmm, weird, I copied and ran the workbook and the assignment was not there. I see it in the commit though.
Disregard this comment please.